tests/functional/models/test_device.py

```diff
@@ -374,6 +374,17 @@ def test_get_supervisor_state(self):
             self.balena.models.device.get_supervisor_state(device['uuid'])
         self.assertIn('No online device(s) found', cm.exception.message)
 
+    def test_get_supervisor_target_state(self):
+        device = self.helper.create_device()[1]
+
+        # should be rejected if the device doesn't exist.
```

Comment:
@xginn8, this test case isn't correct, I've tried and the API will return a 304 with no message-body when try to GET the state of invalid uuid or uuid you don't have access. I think  you should GET the device first in the code with `self.get(uuid)` which will raise `DeviceNotFound` if you don't have access to that uuid and update this case to catch that exception.